python/lib/modeldata/dmod/modeldata/subset/subset_handler.py

```diff
@@ -0,0 +1,243 @@
+        ::class:`Catchment` or ::class:`Nexus` object) for easy lookups.  Return this dictionary once generation is
+        complete.
+
+        Returns
+        -------
+        Dict[str, Union[Catchment, Nexus]]
+            A dictionary of the nodes of the graph keyed by each's string id value.
+        """
+        # Keys of nexus id to lists of catchment ids for the catchments receiving water from this nexus
+        nexus_receiving_cats = dict()
+        # Keys of nexus id to lists of catchment ids for the catchments contributing water to this nexus
+        nexus_contrib_cats = dict()
+        known_catchment_ids = set()
+        known_nexus_ids = set()
+        cat_to = dict()
+        cat_from = dict()
+
+        for cat_id in self._catchment_geo_df.index:
+            known_catchment_ids.add(cat_id)
+            # TODO: do we need to account for more than one downstream?
+            to_nex_id = self._catchment_geo_df.loc[cat_id]['toid'].strip()
+            known_nexus_ids.add(to_nex_id)
+            cat_to[cat_id] = to_nex_id
+            if to_nex_id in nexus_contrib_cats:
+                nexus_contrib_cats[to_nex_id].add(cat_id)
+            else:
+                nexus_contrib_cats[to_nex_id] = {cat_id}
+            # TODO: do we need to account for contained/containing/conjoined?
+        for nex_id in self._nexus_geo_df.index:
+            known_nexus_ids.add(nex_id)
+            to_cats = self._nexus_geo_df.loc[nex_id]['toid']
+            # Handle the first one with conditional check separate, to optimize later ones
+            first_cat_id = to_cats.split(',')[0].strip()
+            if nex_id in nexus_receiving_cats and to_cats:
+                nexus_receiving_cats[nex_id].add(first_cat_id)
+            else:
+                nexus_receiving_cats[nex_id] = {first_cat_id}
+            known_catchment_ids.add(first_cat_id)
+            cat_from[first_cat_id] = nex_id
+            # Now add any remaining
+            for cat_id in to_cats[1:]:
+                clean_cat_id = cat_id.strip()
+                nexus_receiving_cats[nex_id].add(clean_cat_id)
+                known_catchment_ids.add(clean_cat_id)
+                cat_from[cat_id] = nex_id
+
+        hf = dict()
+        # Create the catchments first, just without any upstream/downstream connections
+        for cat_id in known_catchment_ids:
+            # TODO: do params need to be something different?
+            hf[cat_id] = Catchment(catchment_id=cat_id, params=dict())
+        # Create the nexuses next, applying the right collections of catchments for contrib and receiv
+        for nex_id in known_nexus_ids:
+            contributing = set()
+            for cid in nexus_contrib_cats[nex_id]:
+                contributing.add(hf[cid])
+            receiving = set()
+            for cid in nexus_receiving_cats[nex_id]:
+                receiving.add(hf[cid])
+            # TODO: fix as hydro_location should probably not be None
```

Comment:
Can replace None with essentially empty [HydroLocation()](https://github.com/NOAA-OWP/hypy/blob/8220c25ab3f15d2c73f0f7951e9df66017c48aae/python/hypy/hydrolocation/hydrolocation.py#L50)

Reply:
Ah, ok, I'll adjust that ...